File: test-fixtures/basic-project/src/controllers/userController.ts
import { UserService } from "../services/userService";
import { UserRole } from "../models/user";
import { Logger } from "../utils/logger";

export class UserController {
  private userService: UserService;

  constructor() {
    const logger = new Logger("UserController");
    this.userService = new UserService(logger);
  }

  async handleGetUsers(): Promise<string> {
    const users = await this.userService.getAllUsers();
    return JSON.stringify(users);
  }

  async handleGetUser(id: string): Promise<string> {
    const user = await this.userService.getUserById(id);
    if (!user) {
      return JSON.stringify({ error: "Not found" });
    }
    return JSON.stringify(user);
  }

  async handleCreateUser(
    name: string,
    email: string,
    role: string
  ): Promise<string> {
    const userRole = role as UserRole;
    const user = await this.userService.addUser(name, email, userRole);
    return JSON.stringify(user);
  }
}

File: tests/fixtures/java_project/src/main/java/com/example/controller/UserController.java
package com.example.controller;

import com.example.model.User;
import com.example.service.UserService;

public class UserController {
    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    public User handleCreateUser(String name, String email) {
        return userService.createUser(name, email);
    }
}
